pkg/azureutils/azure_disk_utils.go

```diff
@@ -355,14 +355,28 @@ func IsValidDiskURI(diskURI string) error {
 }
 
 func IsValidVolumeCapabilities(volCaps []*csi.VolumeCapability) bool {
+	if ok := IsValidAccessModes(volCaps); !ok {
+		return false
+	}
+	for _, c := range volCaps {
+		blockVolume := c.GetBlock()
+		mountVolume := c.GetMount()
+		accessMode := c.GetAccessMode().GetMode()
+
+		if (blockVolume == nil && mountVolume == nil) ||
+			(blockVolume != nil && mountVolume != nil) {
+			return false
+		}
+		if mountVolume != nil && accessMode == csi.VolumeCapability_AccessMode_MULTI_NODE_MULTI_WRITER {
```

Comment:
for file system, it does not support following 3 access modes:
```
	VolumeCapability_AccessMode_MULTI_NODE_READER_ONLY VolumeCapability_AccessMode_Mode = 3
	VolumeCapability_AccessMode_MULTI_NODE_SINGLE_WRITER VolumeCapability_AccessMode_Mode = 4
	VolumeCapability_AccessMode_MULTI_NODE_MULTI_WRITER VolumeCapability_AccessMode_Mode = 5
```

Reply:
Oh, good to know! I'll add those as well.